### PR TITLE
test(pod): playwright refactor

### PR DESCRIPTION
**Pull request by ZhuoyuJin**
### Proposed behaviour

Refactor Pod tests from Cypress to Playwright

### Current behaviour

Tests are currently Cypress

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Playwright automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required
- [ ] Related docs have been updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Additional context

N/A

### Testing instructions
- [ ] Run Pod tests in Playwright Test Runner to check if the `pod.pw.tsx` file passed
- [ ] Run all other tests in Playwright Test Runner to check none of the other `*.pw.tsx` files have regressed

### CODE REVIEW

File: src/components/pod/pod.pw.tsx [hunk 1]
```diff
@@ -0,0 +1,619 @@
...
+import { SIZE, CHARACTERS } from "../../../playwright/support/constants";
+import { VariantTypes } from "../typography/typography.component";
+
+const specialCharacters = [CHARACTERS.DIACRITICS, CHARACTERS.SPECIALCHARACTERS];
+
+const variantTypes1: VariantTypes[] = ["h1", "h2", "h3", "h4", "h5"];
+
+const variantTypes2: VariantTypes[] = ["h2", "h3", "p", "em", "b"];
+
+test.describe("Prop tests for Pod", () => {
```
Comment (divyajindel) [inline src/components/pod/pod.pw.tsx:41]:
```suggestion
test.describe("prop checks for Pod component", () => {
```

File: src/components/pod/pod.pw.tsx [hunk 2]
```diff
@@ -0,0 +1,619 @@
...
+  ([
+    [true, "1px", "solid", "rgb(204, 214, 219)"],
+    [false, "0px", "none", "rgba(0, 0, 0, 0.9)"],
+  ] as [PodProps["border"], string, string, string][]).forEach(
+    ([boolVal, px, style, color]) => {
+      test(`should render with ${boolVal} as border and ${px} as border value`, async ({
```
Comment (divyajindel) [inline src/components/pod/pod.pw.tsx:47]:
```suggestion
      test(`should render with border value as ${borderValue} when border prop is ${boolVal}`, async ({
```

File: src/components/pod/pod.pw.tsx [hunk 3]
```diff
@@ -0,0 +1,619 @@
...
+const variantTypes1: VariantTypes[] = ["h1", "h2", "h3", "h4", "h5"];
+
+const variantTypes2: VariantTypes[] = ["h2", "h3", "p", "em", "b"];
+
+test.describe("Prop tests for Pod", () => {
+  ([
+    [true, "1px", "solid", "rgb(204, 214, 219)"],
+    [false, "0px", "none", "rgba(0, 0, 0, 0.9)"],
+  ] as [PodProps["border"], string, string, string][]).forEach(
+    ([boolVal, px, style, color]) => {
```
Comment (divyajindel) [inline src/components/pod/pod.pw.tsx:46]:
```suggestion
    ([boolVal, borderValue, style, color]) => {
```

File: src/components/pod/pod.pw.tsx [hunk 4]
```diff
@@ -0,0 +1,619 @@
...
+      test(`should render with ${boolVal} as border and ${px} as border value`, async ({
+        mount,
+        page,
+      }) => {
+        await mount(<PodExample border={boolVal} />);
+
+        const elem = podBlock(page);
+        await checkCSSOutline(elem, px, "border", style, color);
```
Comment (divyajindel) [inline src/components/pod/pod.pw.tsx:54]:
```suggestion
        await checkCSSOutline(elem, borderValue, "border", style, color);
```

File: src/components/pod/pod.pw.tsx [hunk 5]
```diff
@@ -0,0 +1,619 @@
...
+
+  ([
+    [SIZE.EXTRASMALL, 120, 66],
+    [SIZE.SMALL, 120, 66],
+    [SIZE.MEDIUM, 136, 82],
+    [SIZE.LARGE, 184, 98],
+    [SIZE.EXTRALARGE, 216, 130],
+  ] as [PodProps["size"], number, number][]).forEach(
+    ([size, expectedHeight, expectedWidth]) => {
+      test(`should render with ${size} as size and ${expectedHeight} as height and ${expectedWidth} as width`, async ({
```
Comment (divyajindel) [inline src/components/pod/pod.pw.tsx:90]:
```suggestion
      test(`should render with ${size} size when height is ${expectedHeight} and width is ${expectedWidth}`, async ({
```

**Review by divyajindel — CHANGES_REQUESTED** [review]

File: src/components/pod/pod.pw.tsx [hunk 6]
```diff
@@ -0,0 +1,619 @@
...
+      page,
+    }) => {
+      const title = "title";
+      await mount(<PodWithVariantTypes title={title} />);
+
+      const variantLocator = page.locator('[data-element="title"]');
+
+      await expect(variantLocator).toContainText(title);
+    });
+  });
```
Comment (divyajindel) [inline src/components/pod/pod.pw.tsx:152]: I'm not sure if variant has been tested here properly. All of the tests looks similar to me so could you please have a look.

File: src/components/pod/pod.pw.tsx [hunk 7]
```diff
@@ -0,0 +1,619 @@
...
+    }) => {
+      const title = "title";
+      const subtitle = "subtitle";
+      await mount(<PodWithVariantTypes title subtitle={subtitle} />);
+
+      const variantLocator = page.locator('[data-element="subtitle"]');
+
+      await expect(variantLocator).toContainText(title);
+    });
+  });
```
Comment (divyajindel) [inline src/components/pod/pod.pw.tsx:178]: I'm not sure if variant has been tested here properly. All of the tests looks similar to me so could you please have a look.

File: src/components/pod/pod.pw.tsx [hunk 8]
```diff
@@ -0,0 +1,619 @@
...
+
+  specialCharacters.forEach((title) => {
+    test(`should render with ${title} as title`, async ({ mount, page }) => {
+      await mount(<PodExample title={title} />);
+
+      await expect(podTitle(page)).toHaveText(title);
+    });
+  });
+
+  variantTypes1.forEach((variantType) => {
```
Comment (Parsium) [inline src/components/pod/pod.pw.tsx:140]: **comment(non-blocking)**: as `variantTypes1` and `variantTypes2` are both only used once, we could alternatively replace these variables with directing passing an array to the `forEach`:

```tsx
(["h1", "h2", "h3", "h4", "h5"] as const).forEach((variantType) => {
  // tests...
})
```

**Review by Parsium — APPROVED** [review]

**Review by divyajindel — APPROVED** [review]

**Pull request merged**